src/Symfony/Component/Serializer/NameConverter/CamelCaseToSnakeCaseNameConverter.php

```diff
@@ -44,19 +44,7 @@ public function __construct(array $attributes = null, $lowerCamelCase = true)
     public function normalize($propertyName)
     {
         if (null === $this->attributes || in_array($propertyName, $this->attributes)) {
-            $lcPropertyName = lcfirst($propertyName);
-            $snakeCasedName = '';
-
-            $len = strlen($lcPropertyName);
-            for ($i = 0; $i < $len; ++$i) {
-                if (ctype_upper($lcPropertyName[$i])) {
-                    $snakeCasedName .= '_'.strtolower($lcPropertyName[$i]);
-                } else {
-                    $snakeCasedName .= strtolower($lcPropertyName[$i]);
-                }
-            }
-
-            return $snakeCasedName;
+            return strtolower(preg_replace('/[A-Z]/', '_\\0', lcfirst($propertyName)));
```

Comment:
this preg_replace doesn't replace those uppercase characters with lowercase ones